flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/cluster/JobManagerLogListHandlerTest.java

```diff
@@ -0,0 +1,119 @@
+import org.apache.flink.runtime.webmonitor.WebMonitorUtils;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.Assert.assertThat;
+
+/**
+ * Test for the {@link JobManagerLogListHandler}.
+ */
+public class JobManagerLogListHandlerTest {
+
+	private static final DispatcherGateway mockRestfulGateway = new TestingDispatcherGateway.Builder().build();
+	private HandlerRequest<EmptyRequestBody, EmptyMessageParameters> handlerRequest;
+	private Configuration config;
+
+	@Rule
+	public TemporaryFolder temporaryFolder = new TemporaryFolder();
+
+	@Before
+	public void setUp() throws Exception {
+		handlerRequest = createRequest();
+		config = new Configuration();
+	}
+
+	@Test
+	public void testGetJobManagerLogsList() throws Exception {
+		List<LogInfo> logsList = Arrays.asList(
+			new LogInfo("jobmanager.log", 0L),
+			new LogInfo("jobmanager.out", 0L),
+			new LogInfo("test.log", 0L));
+		JobManagerLogListHandler jobManagerLogListHandler = createHandler(logsList);
+		LogListInfo logListInfo = jobManagerLogListHandler.handleRequest(handlerRequest, mockRestfulGateway).get();
+		assertThat(logListInfo.getLogInfos(), hasSize(logsList.size()));
+	}
+
+	@Test
+	public void testGetJobManagerLogsListForLogDirIsNull() throws Exception {
+		JobManagerLogListHandler jobManagerLogListHandler = createHandler(Collections.emptyList());
+		LogListInfo logListInfo = jobManagerLogListHandler.handleRequest(handlerRequest, mockRestfulGateway).get();
+		assertThat(logListInfo.getLogInfos(), hasSize(0));
+	}
+
+	private static HandlerRequest<EmptyRequestBody, EmptyMessageParameters> createRequest() throws HandlerRequestException {
+		Map<String, String> pathParameters = Collections.emptyMap();
+		Map<String, List<String>> queryParameters = Collections.emptyMap();
+
+		return new HandlerRequest<>(
+			EmptyRequestBody.getInstance(),
+			EmptyMessageParameters.getInstance(),
+			pathParameters,
+			queryParameters);
+	}
+
+	private JobManagerLogListHandler createHandler(List<LogInfo> logsList) {
+		WebMonitorUtils.LogFileLocation logFileLocation = createLogFileLocation(logsList);
+		return new JobManagerLogListHandler(
+			() -> CompletableFuture.completedFuture(null),
+			TestingUtils.TIMEOUT(),
+			Collections.emptyMap(),
+			JobManagerLogListHeaders.getInstance(),
+			logFileLocation);
+	}
+
+	private WebMonitorUtils.LogFileLocation createLogFileLocation(List<LogInfo> logsList) {
+		try {
+			for (LogInfo logInfo : logsList) {
+				File file = temporaryFolder.newFile(logInfo.getName());
+				config.setString(WebOptions.LOG_PATH, file.getAbsolutePath());
+			}
+		} catch (Exception e) {
+			throw new AssertionError("Could not setup test.", e);
```

Comment:
```suggestion
			throw new RuntimeException("Could not setup test.", e);
```

`AssertionException` might be misleading since we have not asserted anything.